Detectors/Base/include/DetectorsBase/Detector.h

```diff
@@ -69,15 +72,24 @@ class Detector : public FairDetector
                                   Double_t width, Double_t tilt, Double_t lthick = 0., Double_t dthick = 0.,
                                   UInt_t detType = 0, Int_t buildFlag = 0);
 
+    int getMaterial(int imat) const { return (*mMapMaterial.get())[imat]; }
+    int getMedium  (int imed) const { return (*mMapMedium  .get())[imed]; }
+
+    const std::vector<int>& getMapMaterial() const { return *mMapMaterial.get(); }
+    const std::vector<int>& getMapMedium()   const { return *mMapMedium  .get();   }
```

Comment:
Personally I like it if the code is aligned for easy readability:
`    int getMaterial(int imat) const { return (*mMapMaterial.get())[imat]; }
    int getMedium  (int imed) const { return (*mMapMedium  .get())[imed]; }

    const std::vector<int>& getMapMaterial() const { return *mMapMaterial.get(); }
    const std::vector<int>& getMapMedium()   const { return *mMapMedium  .get();   }
`